src/platforms/ios/iosiaphandler.swift

```diff
@@ -0,0 +1,214 @@
+          if await subscription.isEligibleForIntroOffer, let offerType = subscription.introductoryOffer?.type, offerType == .introductory, let introOfferPeriod = subscription.introductoryOffer?.period {
+            trialDays = daysInTrial(for: introOfferPeriod.unit, with: introOfferPeriod.value)
+          } else {
+            trialDays = 0
+          }
+
+          let price = product.displayPrice
+          let currencyCode = product.priceFormatStyle.currencyCode
+
+          let numberOfMonthsInSubscription = monthsInSubscription(for: subscription.subscriptionPeriod.unit, with: subscription.subscriptionPeriod.value)
+          let monthlyPrice: Double = (product.price as NSDecimalNumber).doubleValue / Double(numberOfMonthsInSubscription)
+
+          let numberFormatter = NumberFormatter()
+          numberFormatter.numberStyle = .currency
+          numberFormatter.locale = product.priceFormatStyle.locale
+          guard let monthlyPriceString = numberFormatter.string(from: NSNumber(floatLiteral: monthlyPrice)) else {
+            InAppPurchaseHandler.logger.error(message: "Monthly price could not be converted to a string with numberFormatter: \(monthlyPrice)")
+            assertionFailure()
+            continue
+          }
+
+          productRegistrationCallback(NSString(string: productIdentifier), NSString(string: currencyCode), NSString(string: price), NSString(string: monthlyPriceString), monthlyPrice, trialDays)
+        }
+
+        registrationCompleteCallback()
+
+        } catch {
+          InAppPurchaseHandler.logger.error(message: "Failed product request from the App Store server. \(error)")
+          registrationFailureCallback()
+          assertionFailure()
+        }
+    }
+
+  // MARK: - Starting subscription
+  @objc func startSubscription(for productIdentifier: String) async {
+    guard let product = productList.first(where: {$0.id == productIdentifier}) else {
+      InAppPurchaseHandler.logger.error(message: "Could not find a product with ID \(productIdentifier)")
+      errorCallback()
+      assertionFailure()
+      return
+    }
+
+    do {
+      let purchaseResult = try await product.purchase()
+      await handlePurchaseResult(purchaseResult)
+    } catch {
+      InAppPurchaseHandler.logger.error(message: "Error purchasing product: \(error.localizedDescription)")
+      errorCallback()
+      assertionFailure()
+    }
+  }
+
+  private func handlePurchaseResult(_ purchaseResult: Product.PurchaseResult) async {
+    switch purchaseResult {
+    case .success(let verificationResult):
+      switch verificationResult {
+      case .verified(let transaction):
+        InAppPurchaseHandler.logger.info(message: "StoreKit subscription returned success and verified")
+        let originalTransactionIdentifier: String = "\(transaction.originalID)"
+        successCallback(NSString(string: transaction.productID), NSString(string: originalTransactionIdentifier))
+        await transaction.finish()
```

Comment:
This may just be more Swiftignorance, but do we need a break here? This is the only `case` that doesn't have one.

Reply:
`break`s aren't needed in Swift, and this was a byproduct of my doing too much C++ lately and not thinking about it clearly. I've removed them all here.